PyU4V/replication.py

```diff
@@ -1062,3 +1066,267 @@ def delete_storage_group_srdf(
+                                 'directorId': dir_id, 'portNumber': port_no})
+
+        payload = {
+            'local_ports': local_ports,
+            'remote_rdfg_number': remote_rdfg_number,
+            'remote_ports': remote_ports,
+            'label': label,
+            'local_rdfg_number': local_rdfg_number}
+
+        self.create_resource(category=REPLICATION,
+                             resource_level=SYMMETRIX,
+                             resource_level_id=array_id,
+                             resource_type=RDF_GROUP, payload=payload)
+
+    def modify_rdf_group(self, action, srdf_group_number, array_id=None,
+                         port_list=None, label=None, dev_list=None,
+                         target_rdf_group=None, consistency_exempt=None):
+        """Function to Modify Ports, devices or change label of RDF group.
+
+        Function can be used to Add ports, move volumes between rdf groups,
+        remove ports or rename RDF group.
+
+        :param action: add_ports, remove_ports, move
+        :param srdf_group_number: srdf group number
+                                  for action on local array: int
+        :param array_id: 12 digit serial of array -- str
+        :param port_list: list of ports to be added or removed e.g.
+                         [RF-1E:10, RF-2E:10] --list
+        :param label: Label for group up to 10 characters -- str
+        :param dev_list: list of volumes to be moved between RDF groups -- list
+        :param target_srdf_group: rdfg group to move volumes to -- int
+        :param consistency_exempt: ignore device for consistency checks--bool
+        """
+        rdfg_actions = {'MOVE': 'Move', 'ADD_PORTS': 'add_ports',
+                        'REMOVE_PORTS': 'remove_ports',
+                        'SET_LABEL': 'set_label'}
+        rdfg_action = rdfg_actions.get(action.upper())
+
+        if not rdfg_action:
+            msg = ('RDFG Action must be one [add_ports, remove_ports, move, '
+                   'set_label]')
+            LOG.exception(msg)
+            raise exception.VolumeBackendAPIException(data=msg)
+        if not array_id:
+            array_id = self.array_id
+        if rdfg_action == 'Move':
+            if not consistency_exempt:
+                consistency_exempt = False
+            payload = {
+                'move': {
+                    'targetRdfGroup': target_rdf_group,
+                    'volumesToMove': dev_list,
+                    'exempt': consistency_exempt
+                },
+                'action': 'Move'}
+
+        elif rdfg_action == 'set_label':
+            payload = {
+                "set_label": {
+                    "label": label
+                },
+                "action": "set_label"
+            }
+
+        elif rdfg_action in ['add_ports', 'remove_ports']:
+            if not port_list:
+                msg = ('list of ports must be supplied when adding or removing'
```

Comment:
there is no space at the end of the first line so when printed to logs or screen it will be 'removingports' instead of 'removing ports'